File: Dataset/OOP/Java/SubClasses.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Subclasses;

/**
 *
 * @author GIA KINH
 */
public class Person {
    private String name;
    private String address;

    public Person() {
    }

    public Person(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Override
    public String toString() {
        return "Person[" + "name=" + name + ",address=" + address + ']';
    }
}


public class Staff extends Person{
    private String school;
    private double pay;

    public Staff(String name, String address, String school, double pay) {
        super(name, address);
        this.school = school;
        this.pay = pay;
    }

    public String getSchool() {
        return school;
    }

    public void setSchool(String school) {
        this.school = school;
    }

    public double getPay() {
        return pay;
    }

    public void setPay(double pay) {
        this.pay = pay;
    }

    @Override
    public String toString() {
        return "Staff[" + super.toString() + "school=" + school + ", pay=" + pay + ']';
    }
    
    
}


public class Student extends Person{
    private String program;
    private int year;
    private double fee;

    public Student(String name, String address, String program, int year, double fee) {
        super(name, address);
        this.program = program;
        this.year = year;
        this.fee = fee;
    }
    
    public Student(String program, int year, double fee, String name, String address) {
        super(name, address);
        this.program = program;
        this.year = year;
        this.fee = fee;
    }


    public String getProgram() {
        return program;
    }

    public void setProgram(String program) {
        this.program = program;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public double getFee() {
        return fee;
    }

    public void setFee(double fee) {
        this.fee = fee;
    }

    @Override
    public String toString() {
        return "Student[" + super.toString() + ",program=" + program + ",year=" + year + ",fee=" + fee + ']';
    }
}


public class TestMain {

    public static void main(String[] args) {
        Student s1 = new Student("PRO192", 2021, 300, "Ha Gia Kinh", "Ha Noi");
        System.out.println(s1);
        s1.setAddress("Quang Binh");
        System.out.println(s1);
        s1.setFee(300);
        s1.setProgram("MAD101");
        s1.setYear(2022);
        System.out.println("Name is " + s1.getName());
        System.out.println("Adress is " + s1.getAddress());
        System.out.println("Fee is " + s1.getFee());
        System.out.println("Program is " + s1.getProgram());
        System.out.println("Year is " + s1.getYear());
        
        Staff sf1 = new Staff("Dieu Linh", "Viet Nam", "Ba Vi", 250);
        System.out.println(sf1);

        sf1.setAddress("Japan");
        System.out.println(sf1);
        sf1.setPay(300);
        sf1.setSchool("Tokyo Universe");
        System.out.println("Name is " + sf1.getName());
        System.out.println("Adress is " + sf1.getAddress());
        System.out.println("Pay is " + sf1.getPay());
        System.out.println("School is " + sf1.getSchool());
    }
}

File: Dataset/Advanced/Base/OOP/Python/SubClasses.py
class Person:
    def __init__(self, name=None, address=None):
        self._name = name
        self._address = address

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, name):
        self._name = name

    @property
    def address(self):
        return self._address

    @address.setter
    def address(self, address):
        self._address = address

    def __str__(self):
        return f"Person[name={self._name}, address={self._address}]"


class Staff(Person):
    def __init__(self, name, address, school, pay):
        super().__init__(name, address)
        self._school = school
        self._pay = pay

    @property
    def school(self):
        return self._school

    @school.setter
    def school(self, school):
        self._school = school

    @property
    def pay(self):
        return self._pay

    @pay.setter
    def pay(self, pay):
        self._pay = pay

    def __str__(self):
        return f"Staff[{super().__str__()} school={self._school}, pay={self._pay}]"


class Student(Person):
    def __init__(self, name, address, program, year, fee):
        super().__init__(name, address)
        self._program = program
        self._year = year
        self._fee = fee

    @property
    def program(self):
        return self._program

    @program.setter
    def program(self, program):
        self._program = program

    @property
    def year(self):
        return self._year

    @year.setter
    def year(self, year):
        self._year = year

    @property
    def fee(self):
        return self._fee

    @fee.setter
    def fee(self, fee):
        self._fee = fee

    def __str__(self):
        return f"Student[{super().__str__()}, program={self._program}, year={self._year}, fee={self._fee}]"


def main():
    s1 = Student("Ha Gia Kinh", "Ha Noi", "PRO192", 2021, 300)
    print(s1)
    s1.address = "Quang Binh"
    print(s1)
    s1.fee = 300
    s1.program = "MAD101"
    s1.year = 2022
    print(f"Name is {s1.name}")
    print(f"Address is {s1.address}")
    print(f"Fee is {s1.fee}")
    print(f"Program is {s1.program}")
    print(f"Year is {s1.year}")

    sf1 = Staff("Dieu Linh", "Viet Nam", "Ba Vi", 250)
    print(sf1)
    sf1.address = "Japan"
    print(sf1)
    sf1.pay = 300
    sf1.school = "Tokyo Universe"
    print(f"Name is {sf1.name}")
    print(f"Address is {sf1.address}")
    print(f"Pay is {sf1.pay}")
    print(f"School is {sf1.school}")


if __name__ == "__main__":
    main()